src/modules/roc_packet/target_stdio/roc_packet/address_to_str.cpp

```diff
@@ -19,29 +19,63 @@ address_to_str::address_to_str(const Address& addr) {
 
     switch (addr.version()) {
     case 4: {
-        if (!addr.get_ip(buffer_, sizeof(buffer_))) {
+        if (!addr.get_host(buffer_, sizeof(buffer_))) {
             roc_log(LogError, "address to str: can't format ip");
         }
 
-        const size_t blen = strlen(buffer_);
+        size_t blen = strlen(buffer_);
         if (snprintf(buffer_ + blen, sizeof(buffer_) - blen, ":%d", addr.port()) < 0) {
             roc_log(LogError, "address to str: can't format port");
         }
 
+        if (addr.has_miface()) {
+            blen = strlen(buffer_);
```

Comment:
Here and in other places. I don't think we can call strlen() after snprintf() returned error because I'm not sure that the string is guaranteed to be null-terminated in this case.

In case of error, I suggest to return early and replace the entire buffer with something like `<error>`.

Reply:
Done